sormas-ui/src/main/java/de/symeda/sormas/ui/configuration/docgeneration/QuarantineTemplatesGrid.java

```diff
@@ -0,0 +1,106 @@
+package de.symeda.sormas.ui.configuration.docgeneration;
+
+import java.io.ByteArrayInputStream;
+
+import com.vaadin.data.provider.DataProvider;
+import com.vaadin.data.provider.ListDataProvider;
+import com.vaadin.icons.VaadinIcons;
+import com.vaadin.server.FileDownloader;
+import com.vaadin.server.Page;
+import com.vaadin.server.StreamResource;
+import com.vaadin.ui.Button;
+import com.vaadin.ui.Grid;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.Notification;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.docgeneneration.QuarantineOrderFacade;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.utils.ValidationException;
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.VaadinUiUtil;
+
+public class QuarantineTemplatesGrid extends Grid<String> {
+
+	private static final long serialVersionUID = 2589713987152595369L;
+
+	public QuarantineTemplatesGrid() {
+		super(String.class);
+		setSizeFull();
+
+		ListDataProvider<String> dataProvider = DataProvider.fromStream(FacadeProvider.getQuarantineOrderFacade().getAvailableTemplates().stream());
+		setDataProvider(dataProvider);
+
+		removeAllColumns();
+		addColumn(x -> x.toString()).setCaption(I18nProperties.getString(Strings.fileName)).setExpandRatio(1);
+		addComponentColumn(this::buildActionButtons).setCaption(I18nProperties.getCaption(Captions.eventActionsView))
+			.setWidth(100)
+			.setStyleGenerator(item -> "v-align-center");
+
+		setSelectionMode(SelectionMode.NONE);
+	}
+
+	public void reload() {
+		// This is bad practice but it works (unlike refreshAll), and in this case its sufficient
+		setItems(FacadeProvider.getQuarantineOrderFacade().getAvailableTemplates());
+		getDataProvider().refreshAll();
+	}
+
+	private Button buildDeleteButton(String s) {
+		Button deleteButton = ButtonHelper.createIconButton("", VaadinIcons.TRASH, e -> {
+			VaadinUiUtil.showDeleteConfirmationWindow(String.format(I18nProperties.getString(Strings.confirmationDeleteFile), s), () -> {
+				try {
+					FacadeProvider.getQuarantineOrderFacade().deleteQuarantineTemplate(s.toString());
+				} catch (ValidationException ex) {
+					new Notification("header i18n delete failed", "content i18n" + ex.getMessage(), Notification.Type.ERROR_MESSAGE, false)
```

Comment:
We shouldn't show exceptions to the user. They won't understand them and they'll confuse them. Instead please display a proper error message - likely just something like "[...] could not be deleted."

Reply:
Fixed.